src/java/org/apache/cassandra/tools/nodetool/GuardrailsConfigCommand.java

```diff
@@ -59,51 +61,107 @@ public static class GetGuardrailsConfig extends GuardrailsConfigCommand
+                                                       else
+                                                           return snakeCase.replaceAll("_fail_", "_");
+                                                   }));
+
+            allGetters.putAll(thresholds);
+
+            allGetters = allGetters.entrySet()
+                                   .stream()
+                                   .sorted(Map.Entry.comparingByKey())
+                                   .collect(Collectors.toMap(
+                                   Map.Entry::getKey,
+                                   Map.Entry::getValue,
+                                   (e1, e2) -> e1,
+                                   LinkedHashMap::new
+                                   ));
+
+            if (allGetters.isEmpty())
+            {
+                assert guardrailName != null;
+                throw new IllegalStateException(String.format("Guardrail %s not found.", guardrailName));
+            }
 
-            display(probe, allGetters, categoryEnum);
+            display(probe, allGetters, categoryEnum, verbose);
         }
 
         @Override
-        public void addRow(List<InternalRow> bucket, GuardrailsMBean mBean, Method method, String guardrailName) throws Throwable
+        public void addRow(List<InternalRow> bucket, GuardrailsMBean mBean, List<Method> methods, String guardrailName) throws Throwable
         {
-            Class<?> returnType = method.getReturnType();
-            Object value = method.invoke(mBean);
-
-            if (returnType.equals(int.class) || returnType.equals(Integer.class)
-                || returnType.equals(long.class) || returnType.equals(Long.class)
-                || returnType.equals(boolean.class) || returnType.equals(Boolean.class)
-                || returnType.equals(Set.class))
-            {
-                constructRow(bucket, guardrailName, value.toString());
-            }
-            else if (returnType.equals(String.class))
+            List<String> values = new ArrayList<>();
+            for (Method method : methods)
             {
-                if (value == null || value.toString().isEmpty())
-                    constructRow(bucket, guardrailName, "null");
+                Class<?> returnType = method.getReturnType();
+                Object value = method.invoke(mBean);
+
+                if (returnType.equals(int.class) || returnType.equals(Integer.class)
+                    || returnType.equals(long.class) || returnType.equals(Long.class)
+                    || returnType.equals(boolean.class) || returnType.equals(Boolean.class)
+                    || returnType.equals(Set.class))
+                {
+                    values.add(value.toString());
+                }
+                else if (returnType.equals(String.class))
+                {
+                    if (value == null || value.toString().isEmpty())
```

Comment:
Should we return an empty string `""`? `null` can be confusing when the return value is truly empty.

Reply:
null is null, empty string is empty string

Reply:
We display `null` for the empty strings, right? Should we echo `empty` instead?